src/crab/ebpf_domain.cpp

```diff
@@ -108,16 +111,90 @@ static std::vector<linear_constraint_t> bit_jmp_to_cst_interval(const NumAbsDoma
     default: throw std::exception();
     }
 
-    return {(result) ? linear_constraint_t::FALSE().negate() : linear_constraint_t::FALSE()};
+    return {(result) ? linear_constraint_t::TRUE() : linear_constraint_t::FALSE()};
+}
+
+static std::vector<linear_constraint_t> signed_jmp_to_cst_interval(const NumAbsDomain& inv, Condition::Op op,
+                                                                   bool is64, variable_t dst_value,
+                                                                   crab::interval_t src_interval) {
+    using namespace crab::dsl_syntax;
+    using Op = Condition::Op;
+
+    // TODO: support a range, but for now we just handle singletons directly here
+    // as needed for the eBPF conformance tests.
+    std::optional<number_t> src_n = src_interval.singleton();
+    if (!src_n || !src_n->fits_cast_to_int64())
+        return {};
+    int64_t src_int_value = src_n.value().cast_to_sint64();
+
+    if (!is64) {
+        auto dst_interval = inv.eval_interval(dst_value);
+        std::optional<number_t> dst_lb = dst_interval.lb().number();
+        std::optional<number_t> dst_ub = dst_interval.ub().number();
+        if (!dst_lb || !dst_lb.value().fits_cast_to_int64() || !dst_ub || !dst_ub.value().fits_cast_to_int64())
+            return {};
+        int32_t dst_lb_value = (int32_t)dst_lb.value().cast_to_sint64();
+        int32_t dst_ub_value = (int32_t)dst_ub.value().cast_to_sint64();
+        src_int_value = (int32_t)src_int_value;
+        switch (op) {
+        case Op::EQ:
+            if (dst_lb_value == src_int_value && dst_ub_value == src_int_value)
+                return {linear_constraint_t::TRUE()};
+            if (dst_lb_value > src_int_value || dst_ub_value < src_int_value)
+                return {linear_constraint_t::FALSE()};
+            return {};
```

Comment:
Returning an empty set and returning TRUE mean the same thing. I assume the choice here is for the sake of symmetry?